cql3/statements/select_statement.cc

```diff
@@ -347,9 +347,10 @@ select_statement::do_execute(service::storage_proxy& proxy,
                                 auto timeout = db::timeout_clock::now() + timeout_duration;
                                 return p->fetch_page(builder, page_size, now, timeout);
                             }
-                    ).then([this, &builder, restrictions_need_filtering] {
+                    ).then([this, p, &builder, restrictions_need_filtering] {
                                 auto rs = builder.build();
                                 if (restrictions_need_filtering) {
+                                    _stats.filtered_rows_read_total += p->stats().rows_read_total;
                                     _stats.filtered_rows_matched_total += rs->size();
```

Comment:
This is (just a tiny bit) icky. How do we know we didn't miss an update?

Perhaps we can just move the definition of pager stats to pager, and add them as a member to cql stats. Then we'd still pass the pager stats to the pager, and be sure we didn't miss an update.

Reply:
I know it's from half a year ago, but I need to refresh this conversation as I don't get the request - how can we miss an update? Which update?

What I meant here is to do two things:
1. Extract pager statistics to a totally separate structure, so we don't poison the API with cql
2. In CQL, use pager statistics to fill our global statistics - note that `filtered_rows_read_total` does not care if the rows were read with paging or not, it just sums all requests. Naturally, it uses value from the pager it used, but it's an implementation detail.

Long time ago, in 3792bed3ed2f8899d801761e7599a64a65782d42, I used a similar trick to pass view stats as cql stats in some places, but in that use case it made a lot of sense - we wanted separate metrics for view writes and normal user writes. Here - I'm pretty sure we don't want separate stats from pagers and non-paged cql requests. Since we don't want separate stats, I saw no reason to incorporate pager stats into cql stats or distinguish between rows read via paging and rows read without it.

So, perhaps I'm missing the point - could you clarify?